### DISCUSSION TITLE
How do I make a Combobox show its first element on startup?

### DISCUSSION THREAD

**Discussion by DrDryg** [AvaloniaUI/Avalonia#12901] [General]
With this code in a xaml file:

```
<ComboBox Height="30" DockPanel.Dock="Top" ItemsSource="{Binding StringData}" SelectedIndex="0">
	<ComboBox.ItemTemplate>
		<DataTemplate x:DataType="x:String">
			<Label Content="{Binding}" FontWeight="Bold"/>
		</DataTemplate>
	</ComboBox.ItemTemplate>
</ComboBox>
```

I dont get the Combobox to show the initial value on startup. How should I achieve the in the best way? I suppose I could put something in the code behind. But it feels like there should be a better way. This must be very common.

**Comment by timunie** [comment]
which Avalonia version?  also how is `StringData` loaded / defined?

**Comment by DrDryg** [comment]
I think I found the problem. I first created my MainWindow and then added my mainViewModel to its context:

```
        public override void OnFrameworkInitializationCompleted()
        {
            if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
            {
                desktop.MainWindow = new MainWindow()
                {
                    DataContext = new MainViewModel()
                };

            }

            base.OnFrameworkInitializationCompleted();
        }
```

Now I set the context in the constructor of MainWindow. Then it works :) Since it didnt have access to the data when being constructed it could not show the value.

Is this the typical way to do it? Settings the contexts through the constructors to ensure that all data is available when constructing the object.

**Reply by timunie** [reply]
Yeah, or use DataTemplates. These will prepare the View depending on the Content-VM set.
